From b03179ead11db7dbfd6a00d3eeef3dac0990f826 Mon Sep 17 00:00:00 2001
From: Sumit Bose <sbose@redhat.com>
Date: Mon, 10 Sep 2018 15:40:14 +0200
Subject: [PATCH 17/83] sbus: dectect python binary for sbus_generate.sh
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

We already detect python2 and python3 binaries during configure. With
this patch PYTHON_EXEC is set to the python3 binary if python3 bindings
are generated and to the python2 binary otherwise. With the help of an
environment variable sbus_generate.sh is made aware of it.

Related to https://pagure.io/SSSD/sssd/issue/3807

Reviewed-by: Pavel Březina <pbrezina@redhat.com>
---
 Makefile.am                             | 4 ++--
 configure.ac                            | 8 ++++++++
 sbus_generate.sh => sbus_generate.sh.in | 2 +-
 3 files changed, 11 insertions(+), 3 deletions(-)
 rename sbus_generate.sh => sbus_generate.sh.in (93%)

diff --git a/Makefile.am b/Makefile.am
index 11d0405..deb9ce3 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1020,14 +1020,14 @@ libsss_cert_la_LDFLAGS = \
     $(NULL)
 
 generate-sbus-code:
-	$(srcdir)/sbus_generate.sh $(abs_srcdir)
+	$(builddir)/sbus_generate.sh $(abs_srcdir)
 
 .PHONY: generate-sbus-code
 
 BUILT_SOURCES += generate-sbus-code
 
 EXTRA_DIST += \
-    sbus_generate.sh \
+    sbus_generate.sh.in \
     src/sbus/codegen/dbus.xml \
     src/sbus/codegen/sbus_CodeGen.py \
     src/sbus/codegen/sbus_DataType.py \
diff --git a/configure.ac b/configure.ac
index 1aac65f..bb18ad4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -373,6 +373,13 @@ them please use argument --without-python3-bindings when running configure.])])
     SSS_CLEAN_PYTHON_VARIABLES
 fi
 
+if test x$HAVE_PYTHON3_BINDINGS = x1; then
+    PYTHON_EXEC=$PYTHON3
+else
+    PYTHON_EXEC=$PYTHON2
+fi
+AC_SUBST(PYTHON_EXEC)
+
 AM_CONDITIONAL([BUILD_PYTHON_BINDINGS],
                [test x"$with_python2_bindings" = xyes \
                      -o x"$with_python3_bindings" = xyes])
@@ -524,4 +531,5 @@ AC_CONFIG_FILES([Makefile contrib/sssd.spec src/examples/rwtab src/doxy.config
                  src/config/setup.py
                  src/systemtap/sssd.stp
                  src/config/SSSDConfig/__init__.py])
+AC_CONFIG_FILES([sbus_generate.sh], [chmod +x sbus_generate.sh])
 AC_OUTPUT
diff --git a/sbus_generate.sh b/sbus_generate.sh.in
similarity index 93%
rename from sbus_generate.sh
rename to sbus_generate.sh.in
index 338fd9d..b2c695e 100755
--- a/sbus_generate.sh
+++ b/sbus_generate.sh.in
@@ -13,7 +13,7 @@ generate() {
 
     echo "Generating sbus code for: $XML"
 
-    python $CODEGEN --sbus sbus --util util \
+    @PYTHON_EXEC@ $CODEGEN --sbus sbus --util util \
         --headers "$HEADERS" \
         --dest "$SRCDIR/src/$DEST" \
         --fileprefix "sbus_${PREFIX}_" \
-- 
2.9.5

